agent-antagonist/src/main.rs

```diff
@@ -60,6 +72,56 @@ enum Args {
         #[clap(short, long)]
         tasks: usize,
     },
+
+    /// In a loop: pick a random region, read a random extent using the repair
+    /// API, and validate that it does not begin with zeroes. Note that this
+    /// will show a false positive if that extent's beginning has never been
+    /// written to.
+    AgentRepairTest {
+        /// Address of the crucible agent
+        #[clap(short, long)]
+        agent: SocketAddr,
+
+        /// Allow reading extents from a read-write downstairs that may not be
+        /// closed.
+        #[clap(short, long)]
+        allow_read_write: bool,
+
+        /// How many bytes to check if they're all zeroes
+        #[clap(long, default_value_t = 32768)]
+        verify_size: usize,
+
+        /// Use Extent::validate
+        #[clap(long)]
+        validate: bool,
+    },
+
+    /// In a loop, pick a random extent for a region, read that extent using the
+    /// repair API, and validate it does not begin with zeroes. Note that this
+    /// will show a false positive if that extent's beginning has never been
+    /// written to.
+    RepairTest {
+        /// Address of the crucible agent
+        #[clap(short, long)]
+        agent: SocketAddr,
+
+        /// Region id
+        #[clap(short, long)]
+        region_id: Uuid,
+
+        /// Allow reading extents from a read-write downstairs that may not be
```

Comment:
This seems fraught with peril.  Can we expect this data to be good?

Reply:
Right, we can't expect it to be good no, if it's accepting IO - but in cases where it's RW downstairs that's not currently doing anything, we should expect the data to be good.